pkg/pool-smart/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -0,0 +1,753 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { BigNumber } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { MinimalSwapInfoPool, TwoTokenPool } from '@balancer-labs/v2-helpers/src/models/vault/pools';
+import { BigNumberish, bn, fp, pct } from '@balancer-labs/v2-helpers/src/numbers';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import LiquidityBootstrappingPool from '@balancer-labs/v2-helpers/src/models/pools/smart/LiquidityBootstrappingPool';
+import { RawLiquidityBootstrappingPoolDeployment } from '@balancer-labs/v2-helpers/src/models/pools/smart/types';
+
+describe('LiquidityBootstrappingPool', function () {
+  let allTokens: TokenList;
+  let trader: SignerWithAddress, recipient: SignerWithAddress;
+  let other: SignerWithAddress, lp: SignerWithAddress, owner: SignerWithAddress;
+
+  const POOL_SWAP_FEE_PERCENTAGE = fp(0.01);
+  const WEIGHTS = [fp(30), fp(70), fp(5), fp(5)];
+  const INITIAL_BALANCES = [fp(0.9), fp(1.8), fp(2.7), fp(3.6)];
+
+  before('setup signers', async () => {
+    [, lp, trader, recipient, other, owner] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy tokens', async () => {
+    allTokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+    await allTokens.mint({ to: [lp, trader], amount: fp(100) });
+  });
+
+  context('for a 1 token pool', () => {
+    it('reverts if there is a single token', async () => {
+      const tokens = await TokenList.create(1);
+      const weights = [fp(1)];
+
+      await expect(LiquidityBootstrappingPool.create({ tokens, weights })).to.be.revertedWith('MIN_TOKENS');
+    });
+  });
+
+  context('for a 2 token pool', () => {
+    itBehavesAsWeightedPool(2);
+  });
+
+  context('for a 3 token pool', () => {
+    itBehavesAsWeightedPool(3);
+  });
```

Comment:
Let's also add a test for a 4 token pool, since that fully test the weight packing logic.

Reply:
Ok; I had that and it seemed to be failing for some other reason - but will add it. Of course it should work with 4 tokens